Plugins/Public/base_plugin/PlayerCommands.cpp

```diff
@@ -1074,179 +1141,180 @@ namespace PlayerCommands
-
-			if (mod->Paused)
+			else if (cmd2 == L"pause")
 			{
-				mod->Paused = false;
-				PrintUserCmdText(client, L"OK Module construction resumed");
+				for (auto& iter = base->modules.begin(); iter != base->modules.end(); iter++)
+				{
+					BuildModule* buildmod = dynamic_cast<BuildModule*>(*iter);
+					if (buildmod && buildmod->active_recipe.nickname == buildRecipe->nickname)
+					{
+						if (!buildmod->Paused)
+						{
+							buildmod->Paused = true;
+							PrintUserCmdText(client, L"Module construction paused");
+						}
+						else
+						{
+							PrintUserCmdText(client, L"ERR Module construction already paused");
+						}
+						return;
+					}
+				}
+				PrintUserCmdText(client, L"ERR Selected module is not being built");
 			}
 			else
 			{
-				PrintUserCmdText(client, L"ERR Module construction is not paused");
+				PrintUserCmdText(client, L"ERR Invalid command");
 			}
-
-			base->Save();
-
 		}
-		else if (cmd == L"construct")
+		else
 		{
-			uint index = ToInt(GetParam(args, ' ', 3));
-			uint type = ToInt(GetParam(args, ' ', 4));
-			if (index < 1 || index >= base->modules.size() || base->modules[index])
-			{
-				PrintUserCmdText(client, L"ERR Module index not valid");
-				return;
-			}
-
-			if (type < Module::TYPE_CORE || type > Module::TYPE_LAST)
-			{
-				PrintUserCmdText(client, L"ERR Module type not available");
-				return;
-			}
-
-			if (type == Module::TYPE_CORE)
-			{
-				if (base->base_level >= 4)
-				{
-					PrintUserCmdText(client, L"ERR Upgrade not available");
-					return;
-				}
-			}
+			PrintUserCmdText(client, L"ERR Invalid parameters");
+			PrintUserCmdText(client, L"/build <list|buildlist> <list|start|resume|pause|info> <moduleName/Nr");
+			PrintUserCmdText(client, L"|  list - lists available module lists");
+			PrintUserCmdText(client, L"|  <buildlist> list - lists modules available on the selected module list");
+			PrintUserCmdText(client, L"|  <buildlist> start <moduleName/Nr> - starts constructon of selected module");
+			PrintUserCmdText(client, L"|  <buildlist> resume <moduleName/Nr> - pauses selected module construction");
+			PrintUserCmdText(client, L"|  <buildlist> pause <moduleName/Nr> - resumes selected module construction");
+			PrintUserCmdText(client, L"|  <buildlist> info <moduleName/Nr> - provides construction material info for selected module");
+		}
+	}
 
-			//make the nickname for inspection
-			uint module_nickname = CreateID(MODULE_TYPE_NICKNAMES[type]);
+	void BaseSwapModule(uint client, const wstring& args)
+	{
+		PlayerBase* base = GetPlayerBaseForClient(client);
+		if (!base)
+		{
+			PrintUserCmdText(client, L"ERR Not in player base");
+			return;
+		}
 
-			if (recipes[module_nickname].reqlevel > base->base_level)
-			{
-				PrintUserCmdText(client, L"ERR Insufficient Core Level");
-				return;
-			}
+		if (!clients[client].admin)
+		{
+			PrintUserCmdText(client, L"ERR Access denied");
+			return;
+		}

```

Comment:
don't we have a function somewhere up above for these two checks?

Reply:
Yes, and I could have sworn I have applied that methoid where applicable a long, long time ago. Applying it once more.